src/azure-cli/azure/cli/command_modules/vm/_params.py

```diff
@@ -677,7 +677,7 @@ def load_arguments(self, _):
         with self.argument_context(scope, arg_group='Network') as c:
             c.argument('vnet_name', help='Name of the virtual network when creating a new one or referencing an existing one.')
             c.argument('vnet_address_prefix', help='The IP address prefix to use when creating a new VNet in CIDR format.')
-            c.argument('subnet', help='The name of the subnet when creating a new VNet or referencing an existing one. Can also reference an existing subnet by ID. If omitted, an appropriate VNet and subnet will be selected automatically, or a new one will be created.')
+            c.argument('subnet', help='The name of the subnet when creating a new VNet or referencing an existing one. Can also reference an existing subnet by ID. If omitted, an appropriate VNet and subnet will be selected automatically only if VNet is also omitted, or a new one will be created if not exists.')
```

Comment:
The word "appropriate" is ambiguous in my mind.

Reply:
Hi @jiasli , could you please help give some suggestions to make this message clearer ?

Reply:
I don't know the behavior of this param. That's why I raised this question. 😆 
We need to state the actual behavior based on the code logic.